docs-src/clusters/how-to-set-up-dual-visibility.md

```diff
@@ -0,0 +1,195 @@
+---
+id: how-to-set-up-dual-visibility
+title: How to set up Dual Visibility
+sidebar_label: Dual Visibility
+description: To enable Dual Visibility, set up a secondary Visibility store with your primary Visibility, and configure your Temporal Cluster to enable read and/or write operations on the secondary Visibility store.
+tags:
+  - operation-guide
+  - filtered-lists
+  - visibility
+ssdi:
+  - Supported from Temporal Server v1.21 onwards.
+---
+
+To enable [Dual Visibility](/concepts/what-is-dual-visibility), set up a secondary Visibility store with your primary Visibility store, and configure your Temporal Cluster to enable read and/or write operations on the secondary Visibility store.
+
+With Dual Visibility, you can read from only one Visibility store at a time, but can configure your Temporal Cluster to write to primary only, secondary only, or to both primary and secondary stores.
+
+#### Set up secondary Visibility store
+
+Set the secondary store with the `secondaryVisibilityStore` configuration key in your Persistence configuration, and then define the secondary Visibility store configuration under `datastores`.
+
+You can configure any of the [supported databases](/cluster-deployment-guide#supported-databases) as your secondary store.
+
+Examples:
+
+To configure MySQL as a secondary store with Cassandra as your primary store, do the following.
+
+```yaml
+persistence:
+  visibilityStore: cass-visibility # This is your primary Visibility store
+  secondaryVisibilityStore: mysql-visibility # This is your secondary Visibility store
+  datastores:
+    cass-visibility:
+      cassandra:
+        hosts: "127.0.0.1"
+        keyspace: "temporal_primary_visibility"
+    mysql-visibility:
+      sql:
+        pluginName: "mysql8" # Verify supported versions. Use a version of SQL that supports advanced Visibility.
+        databaseName: "temporal_secondary_visibility"
+        connectAddr: "127.0.0.1:3306"
+        connectProtocol: "tcp"
+        user: "temporal"
+        password: "temporal"
+```
+
+To configure Elasticsearch as both your primary and secondary store, use the configuration key `elasticsearch.indices.secondary_visibility`, as shown in the following example.
+
+```yaml
+persistence:
+  visibilityStore: es-visibility
+  datastores:
+    es-visibility:
+      elasticsearch:
+        version: "v7"
+        logLevel: "error"
+        url:
+          scheme: "http"
+          host: "127.0.0.1:9200"
+        indices:
+          visibility: temporal_visibility_v1
+          secondary_visibility: temporal_visibility_v2
```

Comment:
The index names needs to be prefixed with `temporal_visibility_v1`. It's kind of weird, but otherwise the template mappings won't be applied automatically.

Maybe something like this:

```suggestion
          visibility: temporal_visibility_v1_dev
          secondary_visibility: temporal_visibility_v1_new
```

Reply:
Is it because the template is called `temporal_visibility_v1_template`? :
`TEMPLATE_URL="${ES_SERVER}/_template/temporal_visibility_v1_template"`

Updated this in the doc

Reply:
No, it's inside the template file, it specifies the template will be applied to any index with name matching `temporal_visibility_v1*`.